app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx

```diff
@@ -129,17 +137,20 @@ export default function Evaluation(props: EvaluationProps) {
     evaluationMaskTargets,
     compareEvaluationMaskTargets,
   ]);
+
   const compareKeys = useMemo(() => {
-    const keys: Record<string, string>[] = [];
+    const currentEval = data?.[`evaluation_${name}`];
+    const currentType = currentEval?.info?.config?.type || "";
+    const currentMethod = currentEval?.info?.config?.method || "";
     const evaluations = data?.evaluations || [];
-    for (const evaluation of evaluations) {
-      const { key, type, method } = evaluation;
-      if (key !== name) {
-        keys.push({ key, type, method });
-      }
-    }
-    return keys;
+    return computeSortedCompareKeys(
```

Comment:
this seems wrong to me. list vs object right?

Reply:
Thanks for catching it.